src/CanvasRenderingContext2d.cc

```diff
@@ -662,10 +663,13 @@ NAN_METHOD(Context2d::PutImageData) {
         uint8_t b = *srcRow++;
         uint8_t a = *srcRow++;
 
-        // argb
-        // performance optimization: fully transparent/opaque pixels can be
-        // processed more efficiently.
-        if (a == 0) {
+        if (dstRow < dstStart) {
```

Comment:
Yes, I agree with you that we should modify the outer code if we can. 

HoweverI'm not very familiar with the algorithm with the strike/row/col, so I just modify to prevent the pointer to write below the memory boundary.

And I think this PR has a bug that will cause the negative x/y not work properly: the other boundary is not considered, and the row/col calculation is not right(sub a negative value). In my code, this caused the crop function does not work as expected.

So finally I drop this method for crop my picture, but it would be nice to prevent the segfault from the code.

BTW: Do you know when will we publish the new version to NPM? 2.0 alpha in the NPM is outdated(so I have to npm install from github master).